compaction/compaction.cc

```diff
@@ -268,15 +268,23 @@ struct compaction_writer {
         : compaction_writer(nullptr, std::move(writer), std::move(sst)) {}
 };
 
+struct splitting_partition_tag { };
+using splitting_partition = bool_class<splitting_partition_tag>;
+
 class compacted_fragments_writer {
     compaction& _c;
     std::optional<compaction_writer> _compaction_writer = {};
-    using creator_func_t = std::function<compaction_writer(const dht::decorated_key&)>;
+    using creator_func_t = std::function<compaction_writer(const dht::decorated_key&, splitting_partition)>;
```

Comment:
Should we do it in compaction, or general sstable writing? Won't the memtable writer and streaming writers also benefit from it?